### PR TITLE
feat: add jwt secret for authentication

### PR DESCRIPTION
**Pull request by johntaiko**

### CODE REVIEW

**Review by smtmfft — APPROVED** [review]
LGTM!
Also, I think the prove-block.sh needs update.
And maybe openAPI doc should take this into account, at lease when auth fails.

File: host/src/lib.rs [hunk 1]
```diff
@@ -106,6 +106,10 @@ pub struct Cli {
     #[serde(flatten)]
     /// Proof request options
     pub proof_request_opt: ProofRequestOpt,
+
+    #[arg(long, require_equals = true)]
```
Comment (smtmfft) [inline host/src/lib.rs:110]: give a default 123456/taiko/whatever maybe easier for script test.

**Review by smtmfft — APPROVED** [review]

File: host/src/lib.rs (hunk 1 above)
Comment (Brechtpd) [reply host/src/lib.rs:110]: yes I think either we need some dummy data so people don't have to set things up to be able to quickly prove blocks, or maybe we could make the jwt check optional by doing `jwt_secret: Option<String>` and only checking it if it's set?

Comment (johntaiko) [reply host/src/lib.rs:110]: Set as `option`, without default value

File: script/prove-block.sh [hunk 2]
```diff
@@ -114,6 +114,7 @@ for block in $(eval echo {$rangeStart..$rangeEnd}); do
 	echo "- proving block $block"
 	curl --location --request POST 'http://localhost:8080/proof' \
 		--header 'Content-Type: application/json' \
+		--header 'Authorization: Bearer 4cbd753fbcbc2639de804f8ce425016a50e0ecd53db00cb5397912e83f5e570e' \
```
Comment (johntaiko) [inline script/prove-block.sh:117]: Add dummy `token` for testing

Comment (johntaiko) [reply script/prove-block.sh:117]:
```sh
cargo run -p raiko-host -- --jwt-secret=4cbd753fbcbc2639de804f8ce425016a50e0ecd53db00cb5397912e83f5e570e
```

**Pull request merged**